prisma/schema.sql

```diff
@@ -0,0 +1,194 @@
+  start_date TIMESTAMPTZ NOT NULL,
+  end_date TIMESTAMPTZ NOT NULL,
+  weekday weekdays NOT NULL,
+  start_time_minutes INTEGER NOT NULL,
+  duration_minutes INTEGER NOT NULL,
+  space_total INTEGER NOT NULL,
+  space_available INTEGER NOT NULL,
+  volunteer_space_total INTEGER NOT NULL,
+  volunteer_space_available INTEGER NOT NULL,
+  is_archived BOOLEAN DEFAULT false NOT NULL,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+-- create parent table
+CREATE TABLE parent (
+  parent_id SERIAL PRIMARY KEY NOT NULL,
+  first_name TEXT NOT NULL,
+  last_name TEXT NOT NULL,
+  phone_number VARCHAR(50) NOT NULL,
+  email TEXT UNIQUE NOT NULL,
+  is_low_income BOOLEAN DEFAULT false,
+  address_line1 TEXT NOT NULL,
+  address_line2 TEXT,
+  postal_code VARCHAR(10) NOT NULL,
+  city_name VARCHAR(50) NOT NULL,
+  province provinces NOT NULL,
+  preferred_language locales NOT NULL,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+-- create volunteer table
+CREATE TABLE volunteer (
+  volunteer_id SERIAL PRIMARY KEY NOT NULL,
+  first_name TEXT NOT NULL,
+  last_name TEXT NOT NULL,
+  email TEXT UNIQUE,
+  phone_number VARCHAR(50),
+  is_valid BOOLEAN DEFAULT false,
+  background_form_link TEXT,
+  address_line1 TEXT,
+  address_line2 TEXT,
+  postal_code VARCHAR(10),
+  city_name TEXT,
+  province provinces,
+  preferred_language locales,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+-- create volunteer registration table
+CREATE TABLE volunteer_reg (
+  volunteer_id INTEGER NOT NULL,
+  program_id INTEGER NOT NULL,
+  FOREIGN KEY(volunteer_id) REFERENCES volunteer(volunteer_id),
+  FOREIGN KEY(program_id) REFERENCES program(program_id),
+  is_valid BOOLEAN DEFAULT false,
+  PRIMARY KEY (volunteer_id, program_id),
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+-- create coupon users table
+CREATE TABLE coupon_users (
+  program_id INTEGER NOT NULL,
+  parent_id INTEGER NOT NULL,
+  FOREIGN KEY (parent_id) REFERENCES parent(parent_id) ON DELETE CASCADE,
+  FOREIGN KEY (program_id) REFERENCES program(program_id) ON DELETE CASCADE,
+  coupon_id TEXT,
+  PRIMARY KEY (parent_id, program_id)
+);
+-- create program waitlist table
+CREATE TABLE program_waitlist(
+  program_id INTEGER NOT NULL,
+  parent_id INTEGER NOT NULL,
+  FOREIGN KEY (parent_id) REFERENCES parent(parent_id) ON DELETE CASCADE,
+  FOREIGN KEY (program_id) REFERENCES program(program_id) ON DELETE CASCADE,
+  PRIMARY KEY (parent_id, program_id),
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+-- create student table
+CREATE TABLE student(
+  student_id SERIAL PRIMARY KEY NOT NULL,
```

Comment:
I think for tables with only one id, you can just call it `id`. I noticed some tables have this but some tables with only 1 id also have the table name in the field name too. Let's try to keep it consistent

Reply:
but then if we have to join a lot of tables that all have ids there will be ambiguous reference like you dont know whose id it is
and then we will have to do more renames or "tablename.id"

Reply:
In Prisma, you don't have to actually write "joins". For example, in the screenshot below, in order to access the data of parent, it's accessed similarly like any other field. 

`model ParentOfStudent {
  parent_id  Int
  student_id Int
  created_at DateTime @default(now()) @db.Timestamptz(6)
  updated_at DateTime @default(now()) @db.Timestamptz(6)
  parent     Parent   @relation(fields: [parent_id], references: [parent_id])
  student    Student  @relation(fields: [student_id], references: [student_id])
  @@id([student_id, parent_id])
  @@map("parent_of_students")
}`

Reply:
For example, this is a query where we join another table.

```
return await prisma.event.findUnique({    where: {      id: parseInt(id),    },    include: {      performances: true,    },  });
```